pkg/kindsys/report.go

```diff
@@ -53,47 +55,123 @@ var plannedCoreKinds = []string{
 	"QueryHistory",
 }
 
+type Kind struct {
+	Name     string `json:"name"`
+	Type     string `json:"type"`
+	Maturity string `json:"maturity"`
+}
+
 type KindStateReport struct {
-	Core       []kindsys.CoreStructuredProperties `json:"core"`
-	Raw        []kindsys.RawProperties            `json:"raw"`
-	Composable []kindsys.ComposableProperties     `json:"composable"`
+	Kinds      map[string]Kind      `json:"kinds"`
+	Dimensions map[string]Dimension `json:"dimensions"`
+}
+
+type Dimension map[string]DimensionValue
```

Comment:
The `map` makes it easier to build, but harder to read later, so we might want to change it into an array.